azavea/pfb-network-connectivity
# Feature/awf/django filter neighborhood vis#387

## Overview

Updates the AnalysisJob and Neighborhood endpoints to only return objects with visibility=PUBLIC when a request is made with no auth credentials.

### Demo

Authenticated (5 results):
![screen shot 2017-05-11 at 09 09 00](https://cloud.githubusercontent.com/assets/1818302/25950627/8d9ce0ca-3629-11e7-8bef-9e5b97188424.png)

No authentication (4 results):
![screen shot 2017-05-11 at 09 09 12](https://cloud.githubusercontent.com/assets/1818302/25950636/97766526-3629-11e7-96c8-0e6a36f12716.png)

Same behavior for the AnalysisJobs endpoint

### Notes

Also fixes a crash that would occur when trying to create a Neighborhood via the admin UI, caused by `ValidationError: last_job cannot be blank`, which it can be because its a nullable field and is read-only.

## Testing Instructions

 * Run migrations
 * Create a private Neighborhood via the admin UI
 * Create a new job or two for the new private Neighborhood via the admin UI (not necessary to run them)
 * Log in/log out of the API while viewing:
  - http://localhost:9200/api/neighborhoods/
  - http://localhost:9200/api/analysis_jobs/
* The job counts should change when you log in/out and private jobs should not be visible while logged out

Closes #387
